marimo/_server/sessions.py

```diff
@@ -366,13 +366,18 @@ def put_input(self, text: str) -> None:
 
     def disconnect_consumer(self) -> None:
         """Stop the session consumer but keep the kernel running"""
-        assert (
-            self.session_consumer is not None
-        ), "Expecting a session consumer to pause"
+        if self.session_consumer is None:
+            LOGGER.debug(
+                "Expecting a session consumer to pause, but none found"
+            )
+            return
+
         LOGGER.debug("Disconnecting session consumer")
-        self.session_consumer.on_stop()
-        self.unsubscribe_consumer()
-        self.session_consumer = None
+        try:
+            self.session_consumer.on_stop()
```

Comment:
This sometimes fails with errors deep in the guts of `websockets`. Finally block makes sure the consumer is removed anyway, so users can still work with the notebook by refreshing